File: 2/1_Cuatrimestre/Programacion_y_Diseño_Orientado_a_Objetos/JavaFinal/NapakalakiJavaSinInterfaz/src/NapakalakiGame/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author javigomez
 */
public class Monster {
    private String name;
    private int combatLevel;
    private Prize prize;
    private BadConsequence badConsequence;
    private int levelChangeAgainstCultisPlayer=0;
    Monster(String n, int l, BadConsequence b, Prize p){
        name=n;
        combatLevel=l;
        badConsequence=b;
        prize=p;
    }
    Monster(String n, int l, BadConsequence badConsequence, Prize p, int IC){
        name=n;
        combatLevel=l;
        this.badConsequence=badConsequence;
        prize=p;
        levelChangeAgainstCultisPlayer=IC;
    }
    public String getName(){
        return name;
    }
    public int getCombatLevel(){
        return combatLevel;
    }
    public BadConsequence getBadConsequence(){
        return badConsequence;
    }
    public int getLevelsGained(){
        return prize.getLevel();
    }
    public int getTreasuresGained(){
        return prize.getTreasures();
    }
    public int getCombatLevelAgainstCultisPlayer(){
        return combatLevel+levelChangeAgainstCultisPlayer;
    }
    
    public String toString(){
        return name+ " level->"+combatLevel+" prize->"+prize.toString()+" badConsequence->"+badConsequence.toString();
    }
    public Prize getPrize(){
        return prize;
    }
}

File: 2/1_Cuatrimestre/Programacion_y_Diseño_Orientado_a_Objetos/RubyFinal/NapakalakiRuby/pkg/NapakalakiRuby-0.0.1/lib/Monster.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'Prize.rb'
require_relative 'BadConsequence.rb'
module NapakalakiGame
class Monster
  attr_reader :name, :combatlevel, :prize, :badConsequence
  def initialize(n, l, b, p)
    @name = n
    @combatlevel = l
    @prize = p
    @badConsequence = b
  end
  def getLevelsGained()
    return prize.levels
  end
  def getTreasuresGained()
    return prize.treasures
  end
  
  
  
  
  
  
  
  
  
  def to_s
   "#{@name}, nivel de combate = #{@combatlevel}  " +badConsequence.to_s
  end
end
end
